src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.Specific.cs

```diff
@@ -118,16 +128,42 @@ public void Windows_MessagePipeTransissionMode()
                     Assert.Equal(msg2.Length, len2);
                     Assert.Equal(msg2, received2);
 
-                    int len3 = server.Read(received3, 0, msg1.Length - 1);  // read one less than message
+                    int expectedRead = msg1.Length - 1;
+                    int len3 = server.Read(received3, 0, expectedRead);  // read one less than message
                     Assert.False(server.IsMessageComplete);
-                    Assert.Equal(msg1.Length - 1, len3);
+                    Assert.Equal(expectedRead, len3);
+                    for (int i = 0; i < expectedRead; ++i)
+                    {
+                        Assert.Equal(msg1[i], received3[i]);
+                    }
 
-                    int len4 = server.Read(received3, len3, received3.Length - len3);
+                    expectedRead = msg1.Length - expectedRead;
+                    Assert.Equal(expectedRead, server.Read(received3, len3, expectedRead));
                     Assert.True(server.IsMessageComplete);
-                    Assert.Equal(msg1.Length, len3 + len4);
                     Assert.Equal(msg1, received3);
 
-                    string userName = server.GetImpersonationUserName();    // not sure what to test here that will work in all cases
```

Comment:
Good catch, will move it into separate test and compare with value retrieved from interop call.